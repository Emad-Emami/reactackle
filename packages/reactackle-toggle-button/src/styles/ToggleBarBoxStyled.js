import PropTypes from 'prop-types';
import styled from 'styled-components';
import { extractThemeOrDefault, getValueString } from 'reactackle-core';

const propTypes = {
  labelPosition: PropTypes.oneOf(['left', 'right']),
  disabled: PropTypes.bool,
  /**
   * Define component theme config
   * See https://github.com/styled-components/styled-components/blob/master/docs/theming.md
   * for more information
   */
  theme: PropTypes.object,
};

const defaultProps = {
  labelPosition: 'left',
  disabled: false,
};

/** PROP RECEIVERS */
const positionProps = ({ labelPosition }) => {
  const spacingLeft = labelPosition === 'left' || !labelPosition ? 'auto' : '0',
    spacingRight = labelPosition === 'left' ? '0' : 'auto';

  return `
    left: ${spacingLeft};
    right: ${spacingRight};
  `;
};

const sizeProps = ({ theme: themeFromProvider }) => {
  const theme = extractThemeOrDefault(themeFromProvider);
  const { width } = theme.reactackle.components.toggleButton.toggleBar,
    { size } = theme.reactackle.components.toggleButton.toggleThumb;

  return `
    width: ${getValueString(width)};
    height: ${getValueString(size)};
  `;
};

const disabled = ({ disabled }) =>
  disabled
    ? `
      &,
      & * {
        cursor: default;
      }
    `
    : `
      &,
      & * {
        cursor: pointer;
      }
    `;

/** STYLES */
export const ToggleBarBoxStyled = styled.div`
  position: absolute;
  top: 0;
  ${positionProps};
  ${sizeProps};
  ${disabled};
`;

ToggleBarBoxStyled.propTypes = propTypes;
ToggleBarBoxStyled.defaultProps = defaultProps;
ToggleBarBoxStyled.displayName = 'ToggleBarBoxStyled';
