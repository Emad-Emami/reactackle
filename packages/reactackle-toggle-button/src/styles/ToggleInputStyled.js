'use strict';

import styled from 'styled-components';
import { elementInvisible } from 'reactackle-core';

const propTypes = {};
const defaultProps = {};

/** STYLES */
export const ToggleInputStyled = styled.input`
  ${/* Prettier workarount until issue is not fixed: https://github.com/prettier/prettier/issues/2291 */ ''};
  ${elementInvisible};
`;

ToggleInputStyled.propTypes = propTypes;
ToggleInputStyled.defaultProps = defaultProps;
ToggleInputStyled.displayName = 'ToggleInputStyled';
