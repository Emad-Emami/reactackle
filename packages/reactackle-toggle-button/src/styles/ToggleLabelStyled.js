import PropTypes from 'prop-types';
import styled from 'styled-components';
import { extractThemeOrDefault, getValueString } from 'reactackle-core';

const propTypes = {
  disabled: PropTypes.bool,
  theme: PropTypes.object,
};

const defaultProps = {
  disabled: false,
};

/** PROP RECEIVERS */
const labelSize = ({ theme: themeFromProvider }) => {
  const theme = extractThemeOrDefault(themeFromProvider);
  const { width } = theme.reactackle.components.toggleButton.toggleBar,
    { size } = theme.reactackle.components.toggleButton.toggleThumb;

  return `
    min-width: ${getValueString(width)};
    min-height: ${getValueString(size)};
  `;
};

const disabled = ({ disabled }) =>
  disabled
    ? `
      &,
      & * {
        cursor: default;
      }
    `
    : `
      &,
      & * {
        cursor: pointer;
      }
    `;

/** STYLES */
export const ToggleLabelStyled = styled.label`
  position: relative;
  display: block;
  user-select: none;
  ${labelSize};
  ${disabled};

  &:hover,
  &:focus {
    outline: none;
  }
`;

ToggleLabelStyled.propTypes = propTypes;
ToggleLabelStyled.defaultProps = defaultProps;
ToggleLabelStyled.displayName = 'ToggleLabelStyled';
