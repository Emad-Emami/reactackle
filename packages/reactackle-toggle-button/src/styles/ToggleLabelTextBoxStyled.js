import PropTypes from 'prop-types';
import styled, { css } from 'styled-components';
import { extractThemeOrDefault, getValueString } from 'reactackle-core';

const propTypes = {
  checked: PropTypes.bool,
  disabled: PropTypes.bool,
  labelPosition: PropTypes.oneOf(['left', 'right']),
  /**
   * Define component theme config
   * See https://github.com/styled-components/styled-components/blob/master/docs/theming.md
   * for more information
   */
  theme: PropTypes.object,
};

const defaultProps = {
  checked: false,
  disabled: false,
  labelPosition: 'left',
};

/** PROP RECEIVERS */
const textBase = ({ theme: themeFromProvider, labelPosition }) => {
  const theme = extractThemeOrDefault(themeFromProvider);
  const {
    fontSize,
    lineHeight,
    toggleLabelSpacing,
  } = theme.reactackle.components.toggleButton.label;

  const { width } = theme.reactackle.components.toggleButton.toggleBar,
    thumbSize = theme.reactackle.components.toggleButton.toggleThumb.size;

  const thumbSizeValue = getValueString(thumbSize),
    fontSizeValue = getValueString(fontSize);

  const labelOffsetTop = `calc((${thumbSizeValue} - ${fontSizeValue} * ${lineHeight}) / 2)`;

  const spacingLeft =
    labelPosition === 'left' || !labelPosition
      ? '0'
      : `calc(${getValueString(toggleLabelSpacing)} + ${getValueString(
          width,
        )})`;

  const spacingRight =
    labelPosition === 'left'
      ? `calc(${getValueString(toggleLabelSpacing)} + ${getValueString(width)})`
      : '0';

  return css`
    font-size: ${getValueString(fontSize)};
    line-height: ${lineHeight};
    padding-top: ${labelOffsetTop};
    padding-left: ${spacingLeft};
    padding-right: ${spacingRight};
  `;
};

const textStyle = ({ checked, disabled, theme: themeFromProvider }) => {
  const theme = extractThemeOrDefault(themeFromProvider);
  const componentPath = theme.reactackle.components.toggleButton.label.state;
  let source = null;

  if (!disabled)
    source = checked ? componentPath.checked : componentPath.unchecked;
  else source = componentPath.disabled;

  const { fontColor } = source;

  return `
    color: ${fontColor};
  `;
};

/** STYLES */
export const ToggleLabelTextBoxStyled = styled.span`
  display: flex;
  ${textBase};
  ${textStyle};
`;

ToggleLabelTextBoxStyled.propTypes = propTypes;
ToggleLabelTextBoxStyled.defaultProps = defaultProps;
ToggleLabelTextBoxStyled.displayName = 'ToggleLabelTextBoxStyled';
