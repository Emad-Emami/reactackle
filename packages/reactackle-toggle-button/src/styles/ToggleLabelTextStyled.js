import PropTypes from 'prop-types';
import styled from 'styled-components';
import { extractThemeOrDefault, getValueString } from 'reactackle-core';

const propTypes = {
  tooltip: PropTypes.string,
  /**
   * Define component theme config
   * See https://github.com/styled-components/styled-components/blob/master/docs/theming.md
   * for more information
   */
  theme: PropTypes.object,
};

const defaultProps = {
  tooltip: '',
};

/** PROP RECEIVERS */
const spacing = ({ theme: themeFromProvider, hasTooltip }) => {
  const theme = extractThemeOrDefault(themeFromProvider);
  const {
    labelTooltipSpacing,
  } = theme.reactackle.components.toggleButton.label;

  return hasTooltip
    ? `margin-right: ${getValueString(labelTooltipSpacing)};`
    : '';
};

/** STYLES */
export const ToggleLabelTextStyled = styled.span`
  ${/* Prettier workarount until issue is not fixed: https://github.com/prettier/prettier/issues/2291 */ ''};
  ${spacing};
`;

ToggleLabelTextStyled.propTypes = propTypes;
ToggleLabelTextStyled.defaultProps = defaultProps;
ToggleLabelTextStyled.displayName = 'ToggleLabelStyled';
