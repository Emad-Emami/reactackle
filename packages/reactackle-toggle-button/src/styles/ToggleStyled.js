import PropTypes from 'prop-types';
import styled from 'styled-components';
import { extractThemeOrDefault } from 'reactackle-core';

const propTypes = {
  disabled: PropTypes.bool,
};

const defaultProps = {
  disabled: false,
};

/** PROP RECEIVERS */
const stateProp = ({ disabled }) => disabled && 'cursor: default;';

const base = ({ theme: themeFromProvider }) => {
  const theme = extractThemeOrDefault(themeFromProvider);

  const fontFamily =
    theme.reactackle.fontFamily[theme.reactackle.body.fontFamily];

  return `
    &,
    & *,
    *::after,
    *::before {
      font-family: ${fontFamily};
      box-sizing: border-box;
    }
  `;
};

/** STYLES */
export const ToggleStyled = styled.div`
  ${base};
  ${stateProp};
`;

ToggleStyled.propTypes = propTypes;
ToggleStyled.defaultProps = defaultProps;
ToggleStyled.displayName = 'ToggleStyled';
