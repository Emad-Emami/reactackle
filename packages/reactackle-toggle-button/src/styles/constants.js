import Color from 'color';
import { colorSecondary } from 'reactackle-core';

const toggleConstants = {
  toggleThumb: {
    backgroundColorChecked: colorSecondary,
  },
};

export const toggleBarBackgroundColorChecked = base =>
  Color(toggleConstants.toggleThumb.backgroundColorChecked(base)).lighten(0.54);
