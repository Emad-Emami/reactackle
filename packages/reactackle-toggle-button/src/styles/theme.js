'use strict';

import {
  baseModule,
  doubleBaseModule,
  twoAndHalfBaseModule,
  fourAndHalfBaseModule,
  colorSecondary,
  colorPaletteGrey100,
  colorPaletteGrey200,
  colorPaletteGrey300,
  colorPaletteGrey400,
  fontColorMedium,
  bodyFontColor,
  radiusRounded,
  fontSizeBody,
} from 'reactackle-core';
import { toggleBarBackgroundColorChecked } from './constants';

export default {
  toggleBar: {
    width: fourAndHalfBaseModule,
    height: doubleBaseModule,

    state: {
      checked: {
        backgroundColor: toggleBarBackgroundColorChecked,
      },

      unchecked: {
        backgroundColor: colorPaletteGrey300,
      },

      disabled: {
        backgroundColor: colorPaletteGrey200,
      },
    },
  },

  toggleThumb: {
    size: twoAndHalfBaseModule,
    borderRadius: radiusRounded,

    state: {
      checked: {
        backgroundColor: colorSecondary,
      },

      unchecked: {
        backgroundColor: colorPaletteGrey100,
      },

      disabled: {
        backgroundColor: colorPaletteGrey400,
      },
    },

    hoverOffset: 3,
  },

  label: {
    toggleLabelSpacing: baseModule,
    labelTooltipSpacing: baseModule,
    fontSize: fontSizeBody,
    lineHeight: 1.5, // Only number's accepted

    state: {
      fontColor: bodyFontColor,

      checked: {
        fontColor: bodyFontColor,
      },

      unchecked: {
        fontColor: bodyFontColor,
      },

      disabled: {
        fontColor: fontColorMedium,
      },
    },
  },
};
