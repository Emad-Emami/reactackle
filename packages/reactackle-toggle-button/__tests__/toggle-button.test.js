'use strict';

import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import { ToggleButton } from '../src';
import { ToggleStyled } from '../src/styles/ToggleStyled';
import { ToggleInputStyled } from '../src/styles/ToggleInputStyled';

jest.mock('react-dom');

describe('<ToggleButton/>', () => {
  it('renders correctly with default props', () => {
    const tree = renderer.create(
      <ToggleButton />,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders correctly if disabled', () => {
    const tree = renderer.create(
      <ToggleButton disabled />,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders correctly if checked', () => {
    const tree = renderer.create(
      <ToggleButton checked />,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders correctly if checked and disabled', () => {
    const tree = renderer.create(
      <ToggleButton checked disabled />,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with tooltip and label', () => {
    const tree = renderer.create(
      <ToggleButton label="Foo" tooltip="Bar" />,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with label', () => {
    const tree = renderer.create(
      <ToggleButton label="Foo" />,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with tooltip, label, labelPosition=right', () => {
    const tree = renderer.create(
      <ToggleButton label="Foo" tooltip="Bar" labelPosition="right" />,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with tooltip, label and checked', () => {
    const tree = renderer.create(
      <ToggleButton label="Foo" tooltip="Bar" checked />,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with tooltip, label and disabled', () => {
    const tree = renderer.create(
      <ToggleButton label="Foo" tooltip="Bar" disabled />,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with tooltip, label, disabled and checked', () => {
    const tree = renderer.create(
      <ToggleButton label="Foo" tooltip="Bar" disabled checked />,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders correctly if hovered', () => {
    const component = mount(
      <ToggleButton />,
    );

    const toggle = component.find(ToggleStyled);
    expect(toJson(component)).toMatchSnapshot();
    toggle.simulate('mouseEnter');
    expect(toJson(component)).toMatchSnapshot();
    toggle.simulate('mouseLeave');
    expect(toJson(component)).toMatchSnapshot();
  });

  it('renders correctly if focused', () => {
    const component = mount(
      <ToggleButton />,
    );
    const toggle = component.find(ToggleInputStyled);

    expect(toJson(component)).toMatchSnapshot();
    
    toggle.simulate('focus');
    expect(toJson(component)).toMatchSnapshot();
    
    toggle.simulate('blur');
    expect(toJson(component)).toMatchSnapshot();
  });

  it('renders correctly if handles input change and call onChange', () => {
    const mockFn = jest.fn(),
      component = mount(
        <ToggleButton onChange={mockFn} />,
      );
    const input = component.findWhere(n => n.type() === 'input');
    
    expect(toJson(component)).toMatchSnapshot();
    input.simulate('change', {
      target: {
        checked: true,
      },
    });
    expect(mockFn).toHaveBeenCalledWith({ value: true });
    expect(toJson(component)).toMatchSnapshot();
  });
});
