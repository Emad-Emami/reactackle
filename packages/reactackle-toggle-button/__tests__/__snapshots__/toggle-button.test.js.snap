// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ToggleButton/> renders correctly if checked 1`] = `
<div
  className="ToggleStyled-hpSctc cNwmaA"
  disabled={false}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <label
    className="ToggleLabelStyled-iqJNYz cCwlqu"
    disabled={false}
    htmlFor="toggle-2"
  >
    <div
      className="ToggleBarBoxStyled-cBeZVR heeuOl"
      disabled={false}
    >
      <label
        checked={true}
        className="ToggleBarStyled-kcTwBB jXzuUU"
        disabled={false}
        htmlFor="toggle-2"
      />
      <input
        checked={true}
        className="ToggleInputStyled-cniALy dfMZHO"
        disabled={false}
        id="toggle-2"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        tabIndex={0}
        type="checkbox"
      />
    </div>
  </label>
</div>
`;

exports[`<ToggleButton/> renders correctly if checked and disabled 1`] = `
<div
  className="ToggleStyled-hpSctc hKhcQp"
  disabled={true}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <label
    className="ToggleLabelStyled-iqJNYz NOqCm"
    disabled={true}
    htmlFor="toggle-3"
  >
    <div
      className="ToggleBarBoxStyled-cBeZVR fHkSgk"
      disabled={true}
    >
      <label
        checked={true}
        className="ToggleBarStyled-kcTwBB ihyzJK"
        disabled={true}
        htmlFor="toggle-3"
      />
      <input
        checked={true}
        className="ToggleInputStyled-cniALy dfMZHO"
        disabled={true}
        id="toggle-3"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        tabIndex={0}
        type="checkbox"
      />
    </div>
  </label>
</div>
`;

exports[`<ToggleButton/> renders correctly if disabled 1`] = `
<div
  className="ToggleStyled-hpSctc hKhcQp"
  disabled={true}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <label
    className="ToggleLabelStyled-iqJNYz NOqCm"
    disabled={true}
    htmlFor="toggle-1"
  >
    <div
      className="ToggleBarBoxStyled-cBeZVR fHkSgk"
      disabled={true}
    >
      <label
        checked={false}
        className="ToggleBarStyled-kcTwBB haaxXZ"
        disabled={true}
        htmlFor="toggle-1"
      />
      <input
        checked={false}
        className="ToggleInputStyled-cniALy dfMZHO"
        disabled={true}
        id="toggle-1"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        tabIndex={0}
        type="checkbox"
      />
    </div>
  </label>
</div>
`;

exports[`<ToggleButton/> renders correctly if focused 1`] = `
<div
  className="ToggleStyled-hpSctc cNwmaA"
  disabled={false}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <label
    className="ToggleLabelStyled-iqJNYz cCwlqu"
    disabled={false}
    htmlFor="toggle-11"
  >
    <div
      className="ToggleBarBoxStyled-cBeZVR heeuOl"
      disabled={false}
    >
      <label
        checked={false}
        className="ToggleBarStyled-kcTwBB aFLLg"
        disabled={false}
        htmlFor="toggle-11"
      />
      <input
        checked={false}
        className="ToggleInputStyled-cniALy dfMZHO"
        disabled={false}
        id="toggle-11"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        tabIndex={0}
        type="checkbox"
      />
    </div>
  </label>
</div>
`;

exports[`<ToggleButton/> renders correctly if focused 2`] = `
<div
  className="ToggleStyled-hpSctc cNwmaA"
  disabled={false}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <label
    className="ToggleLabelStyled-iqJNYz cCwlqu"
    disabled={false}
    htmlFor="toggle-11"
  >
    <div
      className="ToggleBarBoxStyled-cBeZVR heeuOl"
      disabled={false}
    >
      <label
        checked={false}
        className="ToggleBarStyled-kcTwBB gTkPlS"
        disabled={false}
        htmlFor="toggle-11"
      />
      <input
        checked={false}
        className="ToggleInputStyled-cniALy dfMZHO"
        disabled={false}
        id="toggle-11"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        tabIndex={0}
        type="checkbox"
      />
    </div>
  </label>
</div>
`;

exports[`<ToggleButton/> renders correctly if focused 3`] = `
<div
  className="ToggleStyled-hpSctc cNwmaA"
  disabled={false}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <label
    className="ToggleLabelStyled-iqJNYz cCwlqu"
    disabled={false}
    htmlFor="toggle-11"
  >
    <div
      className="ToggleBarBoxStyled-cBeZVR heeuOl"
      disabled={false}
    >
      <label
        checked={false}
        className="ToggleBarStyled-kcTwBB aFLLg"
        disabled={false}
        htmlFor="toggle-11"
      />
      <input
        checked={false}
        className="ToggleInputStyled-cniALy dfMZHO"
        disabled={false}
        id="toggle-11"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        tabIndex={0}
        type="checkbox"
      />
    </div>
  </label>
</div>
`;

exports[`<ToggleButton/> renders correctly if handles input change and call onChange 1`] = `
<div
  className="ToggleStyled-hpSctc cNwmaA"
  disabled={false}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <label
    className="ToggleLabelStyled-iqJNYz cCwlqu"
    disabled={false}
    htmlFor="toggle-12"
  >
    <div
      className="ToggleBarBoxStyled-cBeZVR heeuOl"
      disabled={false}
    >
      <label
        checked={true}
        className="ToggleBarStyled-kcTwBB jXzuUU"
        disabled={false}
        htmlFor="toggle-12"
      />
      <input
        checked={true}
        className="ToggleInputStyled-cniALy dfMZHO"
        disabled={false}
        id="toggle-12"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        tabIndex={0}
        type="checkbox"
      />
    </div>
  </label>
</div>
`;

exports[`<ToggleButton/> renders correctly if hovered 1`] = `
<div
  className="ToggleStyled-hpSctc cNwmaA"
  disabled={false}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <label
    className="ToggleLabelStyled-iqJNYz cCwlqu"
    disabled={false}
    htmlFor="toggle-10"
  >
    <div
      className="ToggleBarBoxStyled-cBeZVR heeuOl"
      disabled={false}
    >
      <label
        checked={false}
        className="ToggleBarStyled-kcTwBB aFLLg"
        disabled={false}
        htmlFor="toggle-10"
      />
      <input
        checked={false}
        className="ToggleInputStyled-cniALy dfMZHO"
        disabled={false}
        id="toggle-10"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        tabIndex={0}
        type="checkbox"
      />
    </div>
  </label>
</div>
`;

exports[`<ToggleButton/> renders correctly if hovered 2`] = `
<div
  className="ToggleStyled-hpSctc cNwmaA"
  disabled={false}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <label
    className="ToggleLabelStyled-iqJNYz cCwlqu"
    disabled={false}
    htmlFor="toggle-10"
  >
    <div
      className="ToggleBarBoxStyled-cBeZVR heeuOl"
      disabled={false}
    >
      <label
        checked={false}
        className="ToggleBarStyled-kcTwBB gTkPlS"
        disabled={false}
        htmlFor="toggle-10"
      />
      <input
        checked={false}
        className="ToggleInputStyled-cniALy dfMZHO"
        disabled={false}
        id="toggle-10"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        tabIndex={0}
        type="checkbox"
      />
    </div>
  </label>
</div>
`;

exports[`<ToggleButton/> renders correctly if hovered 3`] = `
<div
  className="ToggleStyled-hpSctc cNwmaA"
  disabled={false}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <label
    className="ToggleLabelStyled-iqJNYz cCwlqu"
    disabled={false}
    htmlFor="toggle-10"
  >
    <div
      className="ToggleBarBoxStyled-cBeZVR heeuOl"
      disabled={false}
    >
      <label
        checked={false}
        className="ToggleBarStyled-kcTwBB aFLLg"
        disabled={false}
        htmlFor="toggle-10"
      />
      <input
        checked={false}
        className="ToggleInputStyled-cniALy dfMZHO"
        disabled={false}
        id="toggle-10"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        tabIndex={0}
        type="checkbox"
      />
    </div>
  </label>
</div>
`;

exports[`<ToggleButton/> renders correctly with default props 1`] = `
<div
  className="ToggleStyled-hpSctc cNwmaA"
  disabled={false}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <label
    className="ToggleLabelStyled-iqJNYz cCwlqu"
    disabled={false}
    htmlFor="toggle-0"
  >
    <div
      className="ToggleBarBoxStyled-cBeZVR heeuOl"
      disabled={false}
    >
      <label
        checked={false}
        className="ToggleBarStyled-kcTwBB aFLLg"
        disabled={false}
        htmlFor="toggle-0"
      />
      <input
        checked={false}
        className="ToggleInputStyled-cniALy dfMZHO"
        disabled={false}
        id="toggle-0"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        tabIndex={0}
        type="checkbox"
      />
    </div>
  </label>
</div>
`;

exports[`<ToggleButton/> renders correctly with label 1`] = `
<div
  className="ToggleStyled-hpSctc cNwmaA"
  disabled={false}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <label
    className="ToggleLabelStyled-iqJNYz cCwlqu"
    disabled={false}
    htmlFor="toggle-5"
  >
    <div
      className="ToggleBarBoxStyled-cBeZVR heeuOl"
      disabled={false}
    >
      <label
        checked={false}
        className="ToggleBarStyled-kcTwBB aFLLg"
        disabled={false}
        htmlFor="toggle-5"
      />
      <input
        checked={false}
        className="ToggleInputStyled-cniALy dfMZHO"
        disabled={false}
        id="toggle-5"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        tabIndex={0}
        type="checkbox"
      />
    </div>
    <span
      checked={false}
      className="ToggleLabelTextBoxStyled-epGXVn XtjXC"
      disabled={false}
    >
      <span
        className="ToggleLabelTextStyled-geBoni jSzCbs"
      >
        Foo
      </span>
    </span>
  </label>
</div>
`;

exports[`<ToggleButton/> renders correctly with tooltip and label 1`] = `
<div
  className="ToggleStyled-hpSctc cNwmaA"
  disabled={false}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <label
    className="ToggleLabelStyled-iqJNYz cCwlqu"
    disabled={false}
    htmlFor="toggle-4"
  >
    <div
      className="ToggleBarBoxStyled-cBeZVR heeuOl"
      disabled={false}
    >
      <label
        checked={false}
        className="ToggleBarStyled-kcTwBB aFLLg"
        disabled={false}
        htmlFor="toggle-4"
      />
      <input
        checked={false}
        className="ToggleInputStyled-cniALy dfMZHO"
        disabled={false}
        id="toggle-4"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        tabIndex={0}
        type="checkbox"
      />
    </div>
    <span
      checked={false}
      className="ToggleLabelTextBoxStyled-epGXVn XtjXC"
      disabled={false}
    >
      <span
        className="ToggleLabelTextStyled-geBoni evgnHk"
      >
        Foo
      </span>
      <span
        className="ToggleTooltipBoxStyled-ehykpg jWsDJG"
      >
        <div
          className="TooltipIconStyled-egvEre kUiAus"
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          tabIndex={0}
        >
          <span>
            ?
          </span>
        </div>
      </span>
    </span>
  </label>
</div>
`;

exports[`<ToggleButton/> renders correctly with tooltip, label and checked 1`] = `
<div
  className="ToggleStyled-hpSctc cNwmaA"
  disabled={false}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <label
    className="ToggleLabelStyled-iqJNYz cCwlqu"
    disabled={false}
    htmlFor="toggle-7"
  >
    <div
      className="ToggleBarBoxStyled-cBeZVR heeuOl"
      disabled={false}
    >
      <label
        checked={true}
        className="ToggleBarStyled-kcTwBB jXzuUU"
        disabled={false}
        htmlFor="toggle-7"
      />
      <input
        checked={true}
        className="ToggleInputStyled-cniALy dfMZHO"
        disabled={false}
        id="toggle-7"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        tabIndex={0}
        type="checkbox"
      />
    </div>
    <span
      checked={true}
      className="ToggleLabelTextBoxStyled-epGXVn XtjXC"
      disabled={false}
    >
      <span
        className="ToggleLabelTextStyled-geBoni evgnHk"
      >
        Foo
      </span>
      <span
        className="ToggleTooltipBoxStyled-ehykpg jWsDJG"
      >
        <div
          className="TooltipIconStyled-egvEre kUiAus"
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          tabIndex={0}
        >
          <span>
            ?
          </span>
        </div>
      </span>
    </span>
  </label>
</div>
`;

exports[`<ToggleButton/> renders correctly with tooltip, label and disabled 1`] = `
<div
  className="ToggleStyled-hpSctc hKhcQp"
  disabled={true}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <label
    className="ToggleLabelStyled-iqJNYz NOqCm"
    disabled={true}
    htmlFor="toggle-8"
  >
    <div
      className="ToggleBarBoxStyled-cBeZVR fHkSgk"
      disabled={true}
    >
      <label
        checked={false}
        className="ToggleBarStyled-kcTwBB haaxXZ"
        disabled={true}
        htmlFor="toggle-8"
      />
      <input
        checked={false}
        className="ToggleInputStyled-cniALy dfMZHO"
        disabled={true}
        id="toggle-8"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        tabIndex={0}
        type="checkbox"
      />
    </div>
    <span
      checked={false}
      className="ToggleLabelTextBoxStyled-epGXVn lnKhrU"
      disabled={true}
    >
      <span
        className="ToggleLabelTextStyled-geBoni evgnHk"
      >
        Foo
      </span>
      <span
        className="ToggleTooltipBoxStyled-ehykpg jWsDJG"
      >
        <div
          className="TooltipIconStyled-egvEre kUiAus"
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          tabIndex={0}
        >
          <span>
            ?
          </span>
        </div>
      </span>
    </span>
  </label>
</div>
`;

exports[`<ToggleButton/> renders correctly with tooltip, label, disabled and checked 1`] = `
<div
  className="ToggleStyled-hpSctc hKhcQp"
  disabled={true}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <label
    className="ToggleLabelStyled-iqJNYz NOqCm"
    disabled={true}
    htmlFor="toggle-9"
  >
    <div
      className="ToggleBarBoxStyled-cBeZVR fHkSgk"
      disabled={true}
    >
      <label
        checked={true}
        className="ToggleBarStyled-kcTwBB ihyzJK"
        disabled={true}
        htmlFor="toggle-9"
      />
      <input
        checked={true}
        className="ToggleInputStyled-cniALy dfMZHO"
        disabled={true}
        id="toggle-9"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        tabIndex={0}
        type="checkbox"
      />
    </div>
    <span
      checked={true}
      className="ToggleLabelTextBoxStyled-epGXVn lnKhrU"
      disabled={true}
    >
      <span
        className="ToggleLabelTextStyled-geBoni evgnHk"
      >
        Foo
      </span>
      <span
        className="ToggleTooltipBoxStyled-ehykpg jWsDJG"
      >
        <div
          className="TooltipIconStyled-egvEre kUiAus"
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          tabIndex={0}
        >
          <span>
            ?
          </span>
        </div>
      </span>
    </span>
  </label>
</div>
`;

exports[`<ToggleButton/> renders correctly with tooltip, label, labelPosition=right 1`] = `
<div
  className="ToggleStyled-hpSctc cNwmaA"
  disabled={false}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <label
    className="ToggleLabelStyled-iqJNYz cCwlqu"
    disabled={false}
    htmlFor="toggle-6"
  >
    <div
      className="ToggleBarBoxStyled-cBeZVR gapsjO"
      disabled={false}
    >
      <label
        checked={false}
        className="ToggleBarStyled-kcTwBB aFLLg"
        disabled={false}
        htmlFor="toggle-6"
      />
      <input
        checked={false}
        className="ToggleInputStyled-cniALy dfMZHO"
        disabled={false}
        id="toggle-6"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        tabIndex={0}
        type="checkbox"
      />
    </div>
    <span
      checked={false}
      className="ToggleLabelTextBoxStyled-epGXVn bvDycJ"
      disabled={false}
    >
      <span
        className="ToggleLabelTextStyled-geBoni evgnHk"
      >
        Foo
      </span>
      <span
        className="ToggleTooltipBoxStyled-ehykpg jWsDJG"
      >
        <div
          className="TooltipIconStyled-egvEre kUiAus"
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          tabIndex={0}
        >
          <span>
            ?
          </span>
        </div>
      </span>
    </span>
  </label>
</div>
`;
